add gcc 7.3.0 and clang 6.0.0 cross platform, binutils 2.30 for linux, edit clang 5.0.1 to conform to #59

cosmicexplorer opened the pull request:
See [the previous PR I made](https://github.com/pantsbuild/binaries/pull/60/) for context -- this one has been updated to conform the format specified by #59 by adding `build-support/bin/${pkg}/${plat}/${arch}/${version}/build.sh` and any appropriate directory symlinks. This adds (all `.tar.gz` archives):

linux and osx:
- `clang`: 6.0.0 (5.0.1 already exists)
- `gcc`: 7.3.0

linux-only:
- `binutils`: 2.30

The rationale for these packages is described in https://github.com/pantsbuild/pants/pull/5490, which consumes them to form a "native toolchain" to compile and link C/C++ code.

stuhood commented on build-support/bin/clang/linux/x86_64/5.0.1/build.sh:
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
```
Do we actually need both of these? For my own sanity, it would be nice to publish the two versions independently/lazily if possible.

cosmicexplorer replied:
These can be done entirely independently, the `build-clang.sh` script at the root directory (and the `build-clang-for-linux-with-cmake.sh` script) can be invoked with either version to produce the correct binary. I'm not sure what you're asking -- pantsbuild/pants#5490 defaults to version `6.0.0`, and the package for version `5.0.1` doesn't need to be updated, so I could just remove this script?

stuhood replied:
Sure.

stuhood commented on utils.bash:
```diff
@@ -0,0 +1,147 @@
+# Not an executable. Can be sourced by other scripts in this repo for some free
+# error checking for common operations. Uses bash-specific features including
+# `local`.
+
```
This is sortof problematic, because the idea of these scripts is to have reproducibility moving forward. If someone were to come along and edit this and break old scripts, we would be fairly hosed trying to rebuild old binaries.

Could we maybe get ahead of that by versioning this? `utils.v1.bash`, and then just committing to (almost) never changing it?

stuhood replied:
In general though, a bit more copypasta is fine here.

cosmicexplorer replied:
What if instead of `source`ing whatever's in the current version of `utils.bash`, a script was to request it from git by SHA? e.g.:
```
PAGER=cat git show 306a85117b67e518513e21439e7ddc4eb1fcab9b:utils.bash > utils-306a85117b67e518513e21439e7ddc4eb1fcab9b.bash
source utils-306a85117b67e518513e21439e7ddc4eb1fcab9b.bash
```

This is only a little more work, and resolves the issue of reproducibility while allowing us to maintain and augment the `utils.bash` script. This could even be done in `build.sh` scripts for individual versions of e.g. clang which invoke the `build-clang.sh` script at the root of the repo, as in:
```
PAGER=cat git show 306a85117b67e518513e21439e7ddc4eb1fcab9b:build-clang.sh > build-clang-306a85117b67e518513e21439e7ddc4eb1fcab9b.sh
chmod +x build-clang-306a85117b67e518513e21439e7ddc4eb1fcab9b.sh
./build-clang-306a85117b67e518513e21439e7ddc4eb1fcab9b.sh osx 6.0.0
```

The path to the right of the colon in `git show <sha>:<script-path>` is always relative to the repo root, so this technique would also mean you could run e.g. `build-support/bin/clang/mac/10.13/6.0.0/build.sh` in the same directory if you wanted.

This technique introduces a dependency on git and on not rewriting the history after commits made to master, which are both reasonable enough assumptions to make, I think? This doesn't seem too awkward to use (especially since it's opt-in), it's definitely portable to all supported platforms, and it allows for perfect reproducibility.

Separately, I think at some point (not now!!) we should probably have some sort of testing for each version of the binaries that we publish, which could potentially be modelled after the kind of testing we perform with the IntelliJ plugin, by [cloning and running tests with multiple different versions of Pants](https://github.com/pantsbuild/intellij-pants-plugin/blob/67c6d81fbeba11869713a0338fdc12e04e2ed6f9/.travis.yml#L40). I think this is probably what we want to aim for in terms of reproducibility as well.

stuhood replied:
Ooor just put a version in the name =)

cosmicexplorer commented:
@stuhood I've removed `build-support/bin/clang/.../build.sh` in favor of the existing `build-clang-5.0.1.sh` that I added in #57. I've also copied `utils.bash`, along with `build-gcc.sh`, `build-clang.sh`, `build-binutils.sh`, and/or `build-clang-for-linux-with-cmake.sh` as necessary adjacent to the `build.sh` script located in the appropriate subdirectory of `build-support/`. Let me know if this is what you were thinking.

stuhood commented on build-support/bin/binutils/linux/x86_64/2.30/build.sh:
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
```
I see the following failure when I attempt to run this one:
```
# cd build-support/bin/binutils/linux/x86_64/2.30/
[root@140426093eab 2.30]# ./build.sh
+ readonly TARGET_PLATFORM=linux BINUTILS_VERSION=2.30
+ TARGET_PLATFORM=linux
+ BINUTILS_VERSION=2.30
+ MAKE_JOBS=2
+ case "$TARGET_PLATFORM" in
++ mkdirp_absolute_path binutils-2.30-linux
++ local -r dir_relpath=binutils-2.30-linux
++ mkdir -p binutils-2.30-linux
++ get_existing_absolute_path binutils-2.30-linux
++ local -r path_arg=binutils-2.30-linux
+++ pwd
++ local -r abs_path=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux
++ [[ ! -e /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux ]]
++ echo /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux
+ with_pushd /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux build_linux
+ local -r dest=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux
+ cmd_line=("${@:2}")
+ local -a cmd_line
+ pushd /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux
/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30
+ build_linux
++ fetch_extract_binutils_source_release
++ local -r extracted_dirname=binutils-2.30
++ local -r archive_filename=binutils-2.30.tar.xz
++ local -r release_url=https://ftpmirror.gnu.org/binutils/binutils-2.30.tar.xz
+++ curl_file_with_fail https://ftpmirror.gnu.org/binutils/binutils-2.30.tar.xz binutils-2.30.tar.xz
+++ local -r url=https://ftpmirror.gnu.org/binutils/binutils-2.30.tar.xz expected_outfile=binutils-2.30.tar.xz
+++ curl -L --fail -O https://ftpmirror.gnu.org/binutils/binutils-2.30.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 19.3M  100 19.3M    0     0  3627k      0  0:00:05  0:00:05 --:--:-- 10.8M
+++ get_existing_absolute_path binutils-2.30.tar.xz
+++ local -r path_arg=binutils-2.30.tar.xz
++++ pwd
+++ local -r abs_path=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-2.30.tar.xz
+++ [[ ! -e /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-2.30.tar.xz ]]
+++ echo /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-2.30.tar.xz
++ local -r downloaded_archive=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-2.30.tar.xz
++ extract_for /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-2.30.tar.xz binutils-2.30
++ local -r archive_path=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-2.30.tar.xz
++ result_path_candidates=("${@:2}")
++ local -a result_path_candidates
++ do_extract /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-2.30.tar.xz
++ local -r archive_path=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-2.30.tar.xz
++ case "$archive_path" in
++ check_cmd_or_err xz
++ local -r cmd_name=xz
++ hash xz
/pantsbuild-binaries/utils.v1.bash: line 33: hash: xz: not found
++ die_here
++ cat
The command 'xz' is required to run this script. You may have to
install it using your operating system's package manager.
++ exit 1
+ local -r source_extracted_abs=
++ mkdirp_absolute_path binutils-build
++ local -r dir_relpath=binutils-build
++ mkdir -p binutils-build
++ get_existing_absolute_path binutils-build
++ local -r path_arg=binutils-build
+++ pwd
++ local -r abs_path=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-build
++ [[ ! -e /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-build ]]
++ echo /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-build
+ local -r build_dir_abs=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-build
++ mkdirp_absolute_path binutils-install
++ local -r dir_relpath=binutils-install
++ mkdir -p binutils-install
++ get_existing_absolute_path binutils-install
++ local -r path_arg=binutils-install
+++ pwd
++ local -r abs_path=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-install
++ [[ ! -e /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-install ]]
++ echo /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-install
+ local -r install_dir_abs=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-install
+ with_pushd /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-build build_binutils_with_configure /configure --prefix=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-install
+ local -r dest=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-build
+ cmd_line=("${@:2}")
+ local -a cmd_line
+ pushd /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-build
/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-build /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux /pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30
+ build_binutils_with_configure /configure --prefix=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-install
+ configure_cmd_line=("$@")
+ local -a configure_cmd_line
+ /configure --prefix=/pantsbuild-binaries/build-support/bin/binutils/linux/x86_64/2.30/binutils-2.30-linux/binutils-install
./build-binutils.sh: line 19: /configure: No such file or directory
cp: cannot stat `': No such file or directory
```

stuhood commented on build-support/bin/clang/linux/x86_64/6.0.0/build.sh:
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+yum -y install xz
+
```
```
# cd /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/
[root@140426093eab 6.0.0]# ./build.sh
Loaded plugins: fastestmirror, ovl
Setting up Install Process
Determining fastest mirrors
 * base: mirror.sjc02.svwh.net
 * extras: ftp.linux.ncsu.edu
 * updates: centos.sonn.com
base                                                                                                                                                                                                                                                     | 3.7 kB     00:00
centos-sclo-rh                                                                                                                                                                                                                                           | 3.0 kB     00:00
centos-sclo-rh/primary_db                                                                                                                                                                                                                                | 2.8 MB     00:00
centos-sclo-sclo                                                                                                                                                                                                                                         | 2.9 kB     00:00
centos-sclo-sclo/primary_db                                                                                                                                                                                                                              | 184 kB     00:00
extras                                                                                                                                                                                                                                                   | 3.4 kB     00:00
extras/primary_db                                                                                                                                                                                                                                        |  30 kB     00:00
updates                                                                                                                                                                                                                                                  | 3.4 kB     00:00
updates/primary_db                                                                                                                                                                                                                                       | 7.0 MB     00:01
Resolving Dependencies
--> Running transaction check
---> Package xz.x86_64 0:4.999.9-0.5.beta.20091007git.el6 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================================================================================================================================================================================================================
 Package                                                  Arch                                                         Version                                                                                 Repository                                                  Size
================================================================================================================================================================================================================================================================================
Installing:
 xz                                                       x86_64                                                       4.999.9-0.5.beta.20091007git.el6                                                        base                                                       137 k

Transaction Summary
================================================================================================================================================================================================================================================================================
Install       1 Package(s)

Total download size: 137 k
Installed size: 477 k
Downloading Packages:
xz-4.999.9-0.5.beta.20091007git.el6.x86_64.rpm                                                                                                                                                                                                           | 137 kB     00:00
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
  Installing : xz-4.999.9-0.5.beta.20091007git.el6.x86_64                                                                                                                                                                                                                   1/1
  Verifying  : xz-4.999.9-0.5.beta.20091007git.el6.x86_64                                                                                                                                                                                                                   1/1

Installed:
  xz.x86_64 0:4.999.9-0.5.beta.20091007git.el6

Complete!
+ readonly CMAKE_VERSION_FOR_CLANG=3.9.5
+ CMAKE_VERSION_FOR_CLANG=3.9.5
+ package_clang 6.0.0
+ local -r clang_version=6.0.0
++ build_cmake_for_clang_bootstrap
+++ ./build-cmake.sh linux 3.9.5
+ readonly 'CMAKE_VERSION_PATTERN=^([0-9]+\.[0-9]+)\.[0-9]+$'
+ CMAKE_VERSION_PATTERN='^([0-9]+\.[0-9]+)\.[0-9]+$'
+ readonly TARGET_PLATFORM=linux CMAKE_VERSION=3.9.5
+ TARGET_PLATFORM=linux
+ CMAKE_VERSION=3.9.5
+ case "$TARGET_PLATFORM" in
++ mkdirp_absolute_path cmake-3.9.5-linux
++ local -r dir_relpath=cmake-3.9.5-linux
++ mkdir -p cmake-3.9.5-linux
++ get_existing_absolute_path cmake-3.9.5-linux
++ local -r path_arg=cmake-3.9.5-linux
+++ pwd
++ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux
++ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux ]]
++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux
+ with_pushd /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux build_linux
+ local -r dest=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux
+ cmd_line=("${@:2}")
+ local -a cmd_line
+ pushd /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux
/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0
+ build_linux
+ local -r cmake_platform_description=cmake-3.9.5-Linux-x86_64
++ fetch_extract_cmake_binary_release cmake-3.9.5-Linux-x86_64
++ local -r extracted_dirname=cmake-3.9.5-Linux-x86_64
+++ extract_minor_version 3.9.5
+++ local -r version_string=3.9.5
+++ echo 3.9.5
+++ sed -re 's#^([0-9]+\.[0-9]+)\.[0-9]+$#\1#g'
++ local -r cmake_minor_version=3.9
++ local -r archive_filename=cmake-3.9.5-Linux-x86_64.tar.gz
++ local -r release_url=https://cmake.org/files/v3.9/cmake-3.9.5-Linux-x86_64.tar.gz
+++ curl_file_with_fail https://cmake.org/files/v3.9/cmake-3.9.5-Linux-x86_64.tar.gz cmake-3.9.5-Linux-x86_64.tar.gz
+++ local -r url=https://cmake.org/files/v3.9/cmake-3.9.5-Linux-x86_64.tar.gz expected_outfile=cmake-3.9.5-Linux-x86_64.tar.gz
+++ curl -L --fail -O https://cmake.org/files/v3.9/cmake-3.9.5-Linux-x86_64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 30.9M  100 30.9M    0     0  1331k      0  0:00:23  0:00:23 --:--:-- 1089k
+++ get_existing_absolute_path cmake-3.9.5-Linux-x86_64.tar.gz
+++ local -r path_arg=cmake-3.9.5-Linux-x86_64.tar.gz
++++ pwd
+++ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64.tar.gz
+++ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64.tar.gz ]]
+++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64.tar.gz
++ local -r downloaded_archive=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64.tar.gz
++ extract_for /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64.tar.gz cmake-3.9.5-Linux-x86_64
++ local -r archive_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64.tar.gz
++ result_path_candidates=("${@:2}")
++ local -a result_path_candidates
++ do_extract /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64.tar.gz
++ local -r archive_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64.tar.gz
++ case "$archive_path" in
++ tar zxf /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64.tar.gz
++ for result_path in '"${result_path_candidates[@]}"'
++ [[ -e cmake-3.9.5-Linux-x86_64 ]]
++ get_existing_absolute_path cmake-3.9.5-Linux-x86_64
++ local -r path_arg=cmake-3.9.5-Linux-x86_64
+++ pwd
++ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64
++ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64 ]]
++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64
++ return 0
+ local -r extracted_dir=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64
+ package_cmake /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64
+ local -r installed_dir_abs=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64
+ with_pushd /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64 create_gz_package cmake bin share
+ local -r dest=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64
+ cmd_line=("${@:2}")
+ local -a cmd_line
+ pushd /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64
/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64 /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0
+ create_gz_package cmake bin share
+ local -r pkg_name=cmake
+ from_paths=("${@:2}")
+ local -a from_paths
+ local -r pkg_archive_name=cmake.tar.gz
+ rm -f cmake.tar.gz
+ [[ 2 -eq 0 ]]
+ tar -czf cmake.tar.gz bin share
+ get_existing_absolute_path cmake.tar.gz
+ local -r path_arg=cmake.tar.gz
++ pwd
+ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64/cmake.tar.gz
+ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64/cmake.tar.gz ]]
+ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64/cmake.tar.gz
+ popd
/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0
+ popd
/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0
++ local -r cmake_linux_archive_abs=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64/cmake.tar.gz
++ local -r workdir=cmake-for-linux-packages
+++ mkdirp_absolute_path cmake-for-linux-packages
+++ local -r dir_relpath=cmake-for-linux-packages
+++ mkdir -p cmake-for-linux-packages
+++ get_existing_absolute_path cmake-for-linux-packages
+++ local -r path_arg=cmake-for-linux-packages
++++ pwd
+++ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages
+++ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages ]]
+++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages
++ with_pushd /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages extract_for /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64/cmake.tar.gz bin/cmake
++ local -r dest=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages
++ cmd_line=("${@:2}")
++ local -a cmd_line
++ pushd /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages
/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0
++ extract_for /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64/cmake.tar.gz bin/cmake
++ local -r archive_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64/cmake.tar.gz
++ result_path_candidates=("${@:2}")
++ local -a result_path_candidates
++ do_extract /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64/cmake.tar.gz
++ local -r archive_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64/cmake.tar.gz
++ case "$archive_path" in
++ tar zxf /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-3.9.5-linux/cmake-3.9.5-Linux-x86_64/cmake.tar.gz
++ for result_path in '"${result_path_candidates[@]}"'
++ [[ -e bin/cmake ]]
++ get_existing_absolute_path bin/cmake
++ local -r path_arg=bin/cmake
+++ pwd
++ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake
++ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake ]]
++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake
++ return 0
++ popd
/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0
+ local -r cmake_linux_bin=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake
+ CMAKE_EXE=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake
+ ./build-clang.sh linux 6.0.0
+ readonly TARGET_PLATFORM=linux LLVM_VERSION=6.0.0
+ TARGET_PLATFORM=linux
+ LLVM_VERSION=6.0.0
+ readonly MAKE_JOBS=2
+ MAKE_JOBS=2
+ case "$TARGET_PLATFORM" in
++ mkdirp_absolute_path clang-llvm-6.0.0-linux
++ local -r dir_relpath=clang-llvm-6.0.0-linux
++ mkdir -p clang-llvm-6.0.0-linux
++ get_existing_absolute_path clang-llvm-6.0.0-linux
++ local -r path_arg=clang-llvm-6.0.0-linux
+++ pwd
++ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux
++ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux ]]
++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux
++ validate_cmake
++ [[ ! -f /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake ]]
++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake
+ with_pushd /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux build_linux /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake
+ local -r dest=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux
+ cmd_line=("${@:2}")
+ local -a cmd_line
+ pushd /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux
/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0
+ build_linux /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake
+ local -r cmake_exe=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake
+ local -r llvm_src_dirname=llvm-6.0.0.src
+ local -r cfe_src_dirname=cfe-6.0.0.src
++ mkdirp_absolute_path clang-llvm-build
++ local -r dir_relpath=clang-llvm-build
++ mkdir -p clang-llvm-build
++ get_existing_absolute_path clang-llvm-build
++ local -r path_arg=clang-llvm-build
+++ pwd
++ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-build
++ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-build ]]
++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-build
+ local -r build_dir_abs=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-build
++ mkdirp_absolute_path clang-llvm-install
++ local -r dir_relpath=clang-llvm-install
++ mkdir -p clang-llvm-install
++ get_existing_absolute_path clang-llvm-install
++ local -r path_arg=clang-llvm-install
+++ pwd
++ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-install
++ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-install ]]
++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-install
+ local -r install_dir_abs=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-install
++ fetch_extract_llvm_source_release llvm-6.0.0.src
++ local -r extracted_dirname=llvm-6.0.0.src
++ local -r archive_filename=llvm-6.0.0.src.tar.xz
++ local -r release_url=https://releases.llvm.org/6.0.0/llvm-6.0.0.src.tar.xz
+++ curl_file_with_fail https://releases.llvm.org/6.0.0/llvm-6.0.0.src.tar.xz llvm-6.0.0.src.tar.xz
+++ local -r url=https://releases.llvm.org/6.0.0/llvm-6.0.0.src.tar.xz expected_outfile=llvm-6.0.0.src.tar.xz
+++ curl -L --fail -O https://releases.llvm.org/6.0.0/llvm-6.0.0.src.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 24.1M  100 24.1M    0     0  10.8M      0  0:00:02  0:00:02 --:--:-- 11.6M
+++ get_existing_absolute_path llvm-6.0.0.src.tar.xz
+++ local -r path_arg=llvm-6.0.0.src.tar.xz
++++ pwd
+++ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src.tar.xz
+++ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src.tar.xz ]]
+++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src.tar.xz
++ local -r downloaded_archive=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src.tar.xz
++ extract_for /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src.tar.xz llvm-6.0.0.src
++ local -r archive_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src.tar.xz
++ result_path_candidates=("${@:2}")
++ local -a result_path_candidates
++ do_extract /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src.tar.xz
++ local -r archive_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src.tar.xz
++ case "$archive_path" in
++ check_cmd_or_err xz
++ local -r cmd_name=xz
++ hash xz
++ tar xf /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src.tar.xz
++ for result_path in '"${result_path_candidates[@]}"'
++ [[ -e llvm-6.0.0.src ]]
++ get_existing_absolute_path llvm-6.0.0.src
++ local -r path_arg=llvm-6.0.0.src
+++ pwd
++ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src
++ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src ]]
++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src
++ return 0
+ local -r llvm_src_extracted_abs=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src
++ fetch_extract_llvm_source_release cfe-6.0.0.src
++ local -r extracted_dirname=cfe-6.0.0.src
++ local -r archive_filename=cfe-6.0.0.src.tar.xz
++ local -r release_url=https://releases.llvm.org/6.0.0/cfe-6.0.0.src.tar.xz
+++ curl_file_with_fail https://releases.llvm.org/6.0.0/cfe-6.0.0.src.tar.xz cfe-6.0.0.src.tar.xz
+++ local -r url=https://releases.llvm.org/6.0.0/cfe-6.0.0.src.tar.xz expected_outfile=cfe-6.0.0.src.tar.xz
+++ curl -L --fail -O https://releases.llvm.org/6.0.0/cfe-6.0.0.src.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 11.4M  100 11.4M    0     0  14.2M      0 --:--:-- --:--:-- --:--:-- 16.9M
+++ get_existing_absolute_path cfe-6.0.0.src.tar.xz
+++ local -r path_arg=cfe-6.0.0.src.tar.xz
++++ pwd
+++ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src.tar.xz
+++ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src.tar.xz ]]
+++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src.tar.xz
++ local -r downloaded_archive=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src.tar.xz
++ extract_for /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src.tar.xz cfe-6.0.0.src
++ local -r archive_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src.tar.xz
++ result_path_candidates=("${@:2}")
++ local -a result_path_candidates
++ do_extract /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src.tar.xz
++ local -r archive_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src.tar.xz
++ case "$archive_path" in
++ check_cmd_or_err xz
++ local -r cmd_name=xz
++ hash xz
++ tar xf /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src.tar.xz
++ for result_path in '"${result_path_candidates[@]}"'
++ [[ -e cfe-6.0.0.src ]]
++ get_existing_absolute_path cfe-6.0.0.src
++ local -r path_arg=cfe-6.0.0.src
+++ pwd
++ local -r abs_path=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src
++ [[ ! -e /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src ]]
++ echo /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src
++ return 0
+ local -r cfe_src_extracted_abs=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src
+ with_pushd /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-build build_llvm_with_cmake /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-install /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src
+ local -r dest=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-build
+ cmd_line=("${@:2}")
+ local -a cmd_line
+ pushd /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-build
/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-build /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0
+ build_llvm_with_cmake /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-install /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src
+ local -r cmake_exe=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake install_dir=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-install cfe_dir=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src llvm_dir=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src
+ /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/cmake-for-linux-packages/bin/cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-install -DLLVM_EXTERNAL_CLANG_SOURCE_DIR=/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/cfe-6.0.0.src -DLLVM_EXTERNAL_PROJECTS=clang -DLLVM_TARGETS_TO_BUILD=X86 '-DBACKEND_PACKAGE_STRING=Pants-packaged LLVM for linux, version 6.0.0' /pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/llvm-6.0.0.src
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- The ASM compiler identification is unknown
-- Didn't find assembler
CMake Error at CMakeLists.txt:45 (project):
  No CMAKE_C_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:45 (project):
  No CMAKE_CXX_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:45 (project):
  No CMAKE_ASM_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "ASM" or the CMake cache entry CMAKE_ASM_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Warning: Did not find file Compiler/-ASM
-- Configuring incomplete, errors occurred!
See also "/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-build/CMakeFiles/CMakeOutput.log".
See also "/pantsbuild-binaries/build-support/bin/clang/linux/x86_64/6.0.0/clang-llvm-6.0.0-linux/clang-llvm-build/CMakeFiles/CMakeError.log".
cp: cannot stat `': No such file or directory
```

stuhood commented:
Merged and synced. Thanks @cosmicexplorer!
